projects/components/src/table/table.component.test.ts

```diff
@@ -0,0 +1,105 @@
+import { ActivatedRoute, convertToParamMap } from '@angular/router';
+import { NavigationService } from '@hypertrace/common';
+import { LetAsyncModule } from '@hypertrace/components';
+import { createHostFactory, mockProvider } from '@ngneat/spectator/jest';
+import { MockComponent } from 'ng-mocks';
+import { EMPTY, of } from 'rxjs';
+import { PaginatorComponent } from '../paginator/paginator.component';
+import { TableComponent } from './table.component';
+
+describe('Table component', () => {
+  const data = [
+    {
+      foo: 'bar'
+    },
+    {
+      foo: 'baz'
+    }
+  ];
+
+  const columns = [
+    {
+      field: 'foo'
+    }
+  ];
+  const createHost = createHostFactory({
+    component: TableComponent,
+    shallow: true,
+    imports: [LetAsyncModule],
+    providers: [
+      mockProvider(NavigationService),
+      mockProvider(ActivatedRoute),
+      mockProvider(ActivatedRoute, {
+        queryParamMap: EMPTY
+      }),
+      mockProvider(NavigationService)
```

Comment:
Should we also mock tooltip/popover here? popover depends on `navigation$` which would be undefined here, hence throwing that console error.

Reply:
popover isn't part of this test, so not required.Beauty of shallow tests.

Reply:
But tooltip is. The console warnings I saw were due to the display of tooltips which in turn depends on popover. If it is being initialized in test then it would likely throw the warning (either now or with a future change). I was thinking we should mock the directive to prevent this?

Reply:
Tooltip's not part of this test either - it's shallow, only things explicitly declared here are part of the test.

Reply:
One of the test I was playing with was still throwing the console warning after I converted it into shallow test. Not sure why, may be I did something wrong. Still learning about shallow tests :)

Reply:
So shallow only says - don't throw if a directive/component is missing. If you mark a test as shallow but still import tooltip, then it's not actually shallow. The situation you describe sounds like it was still importing tooltip - here the only real directive we're using is letasync (since structural directives are a lot more painful to mock)

Reply:
I see.